hazelcast/cluster.py

```diff
@@ -101,7 +102,8 @@ def _reconnect(self):
             self._client.shutdown()
 
     def _connect_to_cluster(self):
-        addresses = get_possible_addresses(self._config.network_config.addresses, self.members)
+        provider_addresses = get_provider_addresses(self._address_providers)
+        addresses = get_possible_addresses(provider_addresses, self.members)
```

Comment:
I think addresses should be updated after each attempt so that you wont miss updates in the cluster therefore it should be in the beginning of the while loop : ` while current_attempt <= attempt_limit:`